src/test_s2c_srv.c

```diff
@@ -853,12 +838,8 @@ void* s2cWriteWorker(void* arg) {
 
   while (secs() < stopTime) {
     // attempt to write random data into the client socket
-    n = raw_write(conn, threadBuff, RECLTH); // TODO avoid snd block
-    // socket interrupted, continue attempting to write
-    if ((n == -1) && (errno == EINTR))
-      continue;
-    if (n < 0)
-      break;  // all data written. Exit
+    n = writen_any(conn, threadBuff, RECLTH); // TODO avoid snd block
+    if (n <= 0) break;  // writen_any has failed unrecoverably
```

Comment:
Previously, writen_any could not return a value less than zero. Now it can. This looks ok.